Library/Homebrew/requirement.rb

```diff
@@ -141,9 +144,9 @@ def mktemp(&block)
   private
 
   def infer_name
-    klass = self.class.name || self.class.to_s
-    klass = klass.sub(/(Dependency|Requirement)$/, "")
-                 .sub(/^(\w+::)*/, "")
+    klass = self.class.name
+    klass = klass&.sub(/(Dependency|Requirement)$/, "")
+                 &.sub(/^(\w+::)*/, "")
     return klass.downcase if klass.present?
```

Comment:
```suggestion
    if (klass = self.class.name.presence)
      return klass.sub(/(Dependency|Requirement)$/, "")
                  .sub(/^(\w+::)*/, "")
                  .downcase
    end
```

Reply:
This does have a subtle behavioural difference where `Something::Requirement` would return an empty string rather than fall back to the cask case.

Not sure if anything actually relies on behaviour currently, but I could make `cask_dependents` rely on that. Given you suggest to make that not inline, I could call it `CaskDependents::Requirement` and it can behave as before. Otherwise we need some alternative way to override the name.

Reply:
Gotcha, fine with me to leave as-is then 👍🏻